src/os/os_freertos.c

```diff
@@ -73,12 +73,11 @@ int oscore_event_wait_until(oscore_event ev, oscore_time desired_usec) {
 		return waketick;
 	}
 
-	oscore_time diff = desired_usec-waketick;
-	TickType_t wait = (TickType_t)(diff / 1000) / portTICK_PERIOD_MS;
-	if (xSemaphoreTake(ev, wait) == pdTRUE) {
-		printf("EVENT INTERRUPT!\n");
-		xSemaphoreGive(ev);
-		return 1; // we got an interrupt
+	// My experience this is more consistent
```

Comment:
Of course, it is busylooping.
Ideally, we would not do this to allow other FreeRTOS threads to run, but for now this is fine.

Reply:
we could make it configurable via define